generators/src/filepaths.ts

```diff
@@ -35,50 +80,116 @@ export interface FilepathsOptions {
-            if (extensionsSet.has(stackedExtensions)) {
-              return true;
+    trigger: (oldToken, newToken) => {
+      const oldLastSlashIndex = oldToken.lastIndexOf("/");
+      const newLastSlashIndex = newToken.lastIndexOf("/");
+      if (oldLastSlashIndex !== newLastSlashIndex) return true;
+      return oldToken.slice(0, oldLastSlashIndex) !== newToken.slice(0, newLastSlashIndex);
+    },
+    getQueryTerm: (token) => token.slice(token.lastIndexOf("/") + 1),
+
+    custom: async (_, executeShellCommand, generatorContext) => {
+      const { isDangerous, currentWorkingDirectory, searchTerm } = generatorContext;
+      const currentInsertedDirectory =
+        getCurrentInsertedDirectory(rootDirectory ?? currentWorkingDirectory, searchTerm) ?? "/";
+
+      try {
+        // Use \ls command to avoid any aliases set for ls.
+        const data = await executeShellCommand("command ls -1ApL", currentInsertedDirectory);
+        const sortedFiles = sortFilesAlphabetically(data.split("\n"), [".DS_Store"]);
+
+        const generatorOutputArray: Fig.TemplateSuggestion[] = [];
+        // Then loop through them and add them to the generatorOutputArray
+        // depending on the template type
+        for (const name of sortedFiles) {
+          if (name) {
+            const templateType = name.endsWith("/") ? "folders" : "filepaths";
+            if (
+              (templateType === "filepaths" && showFolders !== "only") ||
+              (templateType === "folders" && showFolders !== "never")
+            ) {
+              generatorOutputArray.push({
+                type: templateType === "filepaths" ? "file" : "folder",
+                name,
+                insertValue: name,
+                isDangerous,
+                context: { templateType },
+              });
             }
-            i -= 1;
-            // `i` may become -1 which is not a valid index, but the extensionSet check at the beginning is not run in that case,
-            // so the wrong extension is not evaluated
-            stackedExtensions = [suggestionExtensions[i], stackedExtensions].join(".");
-          } while (i >= 0);
+          }
         }
-        return false;
-      });
-      if (!editFileSuggestions && !editFolderSuggestions) return filtered;
-
-      return filtered.map((suggestion) => ({
-        ...suggestion,
-        ...((suggestion.type === "file" ? editFileSuggestions : editFolderSuggestions) || {}),
-      }));
+
+        // Filter suggestions. This takes in the array of suggestions, filters it,
+        // and outputs an array of suggestions
+        return postProcessSuggestions(filterSuggestions(generatorOutputArray));
+      } catch (err) {
+        return [];
+      }
     },
   };
 }
+
+export const folders = Object.assign(
+  () => filepathsFn({ showFolders: "only" }),
+  Object.freeze(filepathsFn({ showFolders: "only" }))
+);
+export const filepaths = Object.assign(filepathsFn, Object.freeze(filepathsFn()));
```

Comment:
Also I would like a feedback on this. It allows `filepaths` and `folders` exports to be used both as function and as constants. 
e.g.
```
args: {
  generators: [folders],
}
// equivalent to 
args: {
  generators: [folders()],
}
// same for filepaths
```